### ISSUE TITLE
Suggest Removing Installation Instructions for `godoc` in Favor of `doc` Instructions

### ISSUE THREAD

**Issue by MeirSedlis** [quii/learn-go-with-tests#691]
[Hello World](https://github.com/quii/learn-go-with-tests/blob/9cc63601c72a3b5f2d8cee8220667e1a063ce173/hello-world.md?plain=1#L124-L130) contains instructions for installing `godocs` but the `doc` ([reference](https://pkg.go.dev/cmd/doc)) command works very similarly and is a faster and more lightweight command which may be better suited for learners. 

I would be happy to create a PR for this if you agree that `doc` is a more appropriate command for this section.

**Comment by quii** [comment]
Go for it!

**Comment by Battlesheepu** [comment]
Since this was never updated and `godoc` is now deprecated (as I also mentioned in https://github.com/quii/learn-go-with-tests/issues/653 )

The package is now replaced by `pkgsite`. Should we go for that replacement instead of `doc`, mainly because of it having very similar functionality (namely, serving docs locally on a http server).

**Comment by Battlesheepu** [comment]
@quii 

Okay - for now, I've managed to replace the deprecated URLs, and that's what my MR's including.

The problem is, I did try to generate the local documentation locally and have stumbled upon issues (main one, the error 424, [which seems to be described here](https://github.com/golang/go/issues/51368), although I can't promise my issue is exactly the same).

Do you mind if we just keep the `godoc` as the 'go-to' solution for the docs and simply mention that `pkgsite` is the official replacement, with a reason for sticking to `godoc` saying something along the lines of 

> While the `godoc` package was deprecated and replaced by `pkgsite`, many find `godoc` easier for generating and browsing locally generated docs. If you'd still like to try `pkgsite`, [take a look at the package's site](https://pkg.go.dev/golang.org/x/pkgsite/cmd/pkgsite).

**Comment by Battlesheepu** [comment]
@quii 

Sorry for the vagueness in the previous comment, which I've noticed only after re-reading it. Let me try again.

It seems that `pkgsite` doesn't play nice with modules without a '.' in their name, as these are treated as `std` libraries, which, for me and for the [related issue's creator,](https://github.com/golang/go/issues/51368)  simply breaks the generated docs. I've found out about this only after generating the docs for myself.

I'd like to retract my suggestion of replacing `godoc` with `pkgsite` - instead, I propose simply adding a note about `godoc` being deprecated and replaced by `pkgsite`, so the situation is clear for the reader. What do you think?

**Comment by zacharysyoung** [comment]
@Battlesheepu, following your link to the issue about modules without a dot (.), I've reached some different conclusions.

From, https://github.com/golang/go/issues/51368, I found this issue, https://github.com/golang/go/issues/32819, and the discussion there very clearly concludes that modules without a dot are (almost entirely) reserved, and that the documentation needed to be updated; and it was: https://github.com/golang/website/commit/3e18747147e17de19023de7a334c5a8508cbdff1.

Both of those issues have been closed.

From all that, I conclude that:
- pkgsite will not some day play nice in the way you hope
- learn-go-with-tests might take an opportunity to explain to readers (fresh and seasoned alike) that creating module names without dots (for whatever reason, including brevity for the sake of teaching) comes with risks

**Comment by quii** [comment]
This was resolved via #799 — hello-world.md now uses `go doc` and `pkgsite`, no mention of `godoc` anymore. Closing.
